dojo/tools/bugcrowd_api/parser.py

```diff
@@ -0,0 +1,227 @@
+                try:
+                    if "://" in entry["attributes"]["bug_url"]:  # is the host full uri?
+                        bug_endpoint = Endpoint.from_uri(
+                            entry["attributes"]["bug_url"].strip()
+                        )
+                        # can raise exception if the host is not valid URL
+                    else:
+                        bug_endpoint = Endpoint.from_uri(
+                            "//" + entry["attributes"]["bug_url"].strip()
+                        )
+                        # can raise exception if there is no way to parse the host
+                except Exception as e:  # We don't want to fail the whole import just for 1 error in the bug_url
+                    logger.error(
+                        "Error parsing bugrcrowd bug_url : {}".format(
+                            entry["attributes"]["bug_url"].strip()
+                        )
+                    )
+                bug_url = entry["attributes"]["bug_url"]
+
+            description = "\n".join(
+                [
+                    entry["attributes"]["description"],
+                    "",
+                    "Bugcrowd details:",
+                    f"- Severity: P{ bugcrowd_severity }",
+                    f"- Bug Url: { bug_url }",
+                    "",
+                    f"Bugcrowd link: {links}",
+                ]
+            )
+            mitigation = entry["attributes"]["remediation_advice"]
+            steps_to_reproduce = entry["attributes"]["description"]
+            unique_id_from_tool = entry["id"]
+
+            finding = Finding(
+                test=test,
+                title=textwrap.shorten(title, width=511, placeholder="..."),
+                date=date,
+                severity=self.convert_severity(bugcrowd_severity),
+                description=description,
+                mitigation=mitigation,
+                steps_to_reproduce=steps_to_reproduce,
+                active=self.is_active(bugcrowd_state),
+                verified=self.is_verified(bugcrowd_state),
+                false_p=self.is_false_p(bugcrowd_state),
+                duplicate=bugcrowd_duplicate,
+                out_of_scope=self.is_out_of_scope(bugcrowd_state),
+                risk_accepted=self.is_risk_accepted(bugcrowd_state),
+                is_mitigated=self.is_mitigated(bugcrowd_state),
+                static_finding=False,
+                dynamic_finding=True,
+                unique_id_from_tool=unique_id_from_tool,
+            )
+
+            try:
+                bug_endpoint.clean()
+                try:
+                    finding.unsaved_endpoints = [bug_endpoint]
+                except Exception as e:
+                    logger.error(
+                        "{} bug url from bugcrowd failed to parse to endpoint, error= {}".format(
+                            str(bug_endpoint), e
+                        )
+                    )
+            except:
```

Comment:
If there is any issue in format, `Endpoint.clean()` it should raise only `ValidationError`.
https://github.com/DefectDojo/django-DefectDojo/blob/dc20ee1bf9f99953b50c610bd9840053e8fd12f1/dojo/models.py#L1426-L1500
It is not best practice to use a bare `except:`
> When catching exceptions, mention specific exceptions whenever possible instead of using a bare `except:` clause

Source: https://peps.python.org/pep-0008/#programming-recommendations

https://youtu.be/hgI0p1zf31k?t=111